wrapanapi/systems/google.py

```diff
@@ -0,0 +1,798 @@
+                    'https://www.googleapis.com/auth/logging.write'
+                ]
+            }],
+
+            # Metadata is readable from the instance and allows you to
+            # pass configuration from deployment scripts to instances.
+            'metadata': {
+                'items': [{
+                    # Startup script is automatically executed by the
+                    # instance upon startup.
+                    'key': 'startup-script',
+                    'value': startup_script
+                }, {
+                    # Every project has a default Cloud Storage bucket that's
+                    # the same name as the project.
+                    'key': 'bucket',
+                    'value': self._project
+                }]
+            },
+            'tags': {
+                'items': ['https-server']
+            }
+        }
+
+        if ssh_key:
+            ssh_keys = {
+                'key': 'ssh-keys',
+                'value': ssh_key
+            }
+            config['metadata']['items'].append(ssh_keys)
+
+        operation = self._instances_api.insert(
+            project=self._project, zone=zone, body=config).execute()
+        wait_for(
+            lambda: self.system.is_zone_operation_done(operation['name']), delay=0.5,
+            num_sec=timeout, message=" Create {}".format(instance_name)
+        )
+        instance = GCEInstance(system=self.system, name=instance_name, zone=zone)
+        wait_for(
+            lambda: instance.in_steady_state, timeout=timeout,
+            delay=0.5, message="Instance {} to reach steady state".format(instance_name)
+        )
+        return instance
+
+
+class GoogleCloudSystem(System, TemplateMixin, VmMixin):
+    """
+    Client to Google Cloud Platform API
+
+    """
+    _stats_available = {
+        'num_vm': lambda self: len(self.list_vms()),
+        'num_template': lambda self: len(self.list_templates()),
+    }
+
+    can_suspend = False
+    can_pause = False
+
+    default_scope = ['https://www.googleapis.com/auth/cloud-platform']
+
+    def __init__(self, project=None, zone=None, file_type=None, **kwargs):
+        """
+            The last three argumets are optional and required only if you want
+            to use json or p12 files.
+            By default, we expecting that service_account arg contains service account data.
+
+            Args:
+                project: name of the project, so called project_id
+                zone: zone of cloud
+                service_account: service_account_content
+
+                scope: compute engine, container engine, sqlservice end etc
+                file_path: path to json or p12 file
+                file_type: p12 or json
+                client_email: Require for p12 file
+
+            Returns: A :py:class:`GoogleCloudSystem` object.
+        """
+        super(GoogleCloudSystem, self).__init__(**kwargs)
+        self._project = project
+        self._zone = zone
```

Comment:
BTW - GCE on MIQ/CMFE  doesn't need zone anymore. it was kinda useless - adding a zone - we still were getting access to all regions/zones. so it got totally removed from UI

Reply:
Hmm.. we'll probably take care of this in another PR then :)

Reply:
definetly - not this time. your PR is awesome already and GoogleCloudInstance already has it's own `zone` attr. which would fix few flaky rest-tests